Docs: Another solution for the "no space left on device"

gustavohenrique opened the pull request:
The another solution is editing the ~/boot2docker/.profile and change the DiskSize key.

Commit:
Removal of the regex to replace ips

Signed-off-by: Mingzhen Feng <fmzhen@zju.edu.cn>
Signed-off-by: Gustavo Henrique <gustavo@gustavohenrique.net>

LK4D4 committed:
Use goroutine-safe version of rand.Source

Signed-off-by: Alexander Morozov <lk4d4@docker.com>
Signed-off-by: Gustavo Henrique <gustavo@gustavohenrique.net>

gustavohenrique committed:
Another solution for the "no space left on device"

The another solution is editing the ~/boot2docker/.profile and change the DiskSize key.

Signed-off-by: Gustavo Henrique <gustavo@gustavohenrique.net>

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "patch-1" git@github.com:gustavohenrique/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

This will update the existing PR, so **DO NOT** open a new one.

SvenDowideit commented:
very thumbs up, thankyou @gustavohenrique for this

LGTM - @docker/docs-owners

fredlf commented on docs/sources/articles/b2d_volume_resize.md:
```diff
@@ -6,21 +6,40 @@ page_keywords: boot2docker, volume, virtualbox
+on device" errors when the Boot2Docker volume fills up. There are two solutions
+in this case:
+
+## Solution 1: Fast, but it will delete all local docker data
```
Docker

fredlf commented on docs/sources/articles/b2d_volume_resize.md:
```diff
@@ -6,21 +6,40 @@ page_keywords: boot2docker, volume, virtualbox
+already there):
+
+    # Disk image size in MB
+    DiskSize = 50000
```
There's nothing magic about this allocation, right? It could be whatever is needed?

fredlf commented on docs/sources/articles/b2d_volume_resize.md:
```diff
@@ -6,21 +6,40 @@ page_keywords: boot2docker, volume, virtualbox
+
+## Solution 2: Increase the volume size
+
+The solution increase the volume size by first cloning it, then resizing it using a disk 
```
Missing word?

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "patch-1" git@github.com:gustavohenrique/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~2
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

This will update the existing PR, so **DO NOT** open a new one.

fredlf commented on docs/sources/articles/b2d_volume_resize.md:
```diff
@@ -6,21 +6,39 @@ page_keywords: boot2docker, volume, virtualbox
+
+## Solution 2: Increase the volume size
+
+The solution increase the volume size by first cloning it, then resizing it using a disk partitioning tool. 
```
There's still a word or more missing here. Should it read "The solution is to increase the volume size..."? Don't want to hold up this PR, however, so we can always fix it in a follow-on.

gustavohenrique replied:
@fredlf What do you think about "Increase the size of boot2docker volume"?

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "patch-1" git@github.com:gustavohenrique/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~3
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

This will update the existing PR, so **DO NOT** open a new one.

icecrime commented:
Ping @moxiegirl!

moxiegirl commented on docs/sources/articles/b2d_volume_resize.md:
```diff
@@ -6,21 +6,39 @@ page_keywords: boot2docker, volume, virtualbox
-on device" errors when the Boot2Docker volume fills up. The solution is to 
-increase the volume size by first cloning it, then resizing it using a disk 
-partitioning tool. 
+on device" errors when the Boot2Docker volume fills up. There are two solutions
```
Thank you for the contribution. There are a few problems in the file in terms of structure and flow. We don't ask the user to first output the config with `boot2docker config` which would help the user orient to what to they are working with. 

Also, we should not use `###` to mark steps in a list.  It looks like some of these issues predated your change. So, would you be willing to correct them? If not I can carry.

---> https://gist.github.com/moxiegirl/78e3d1bf244e29346d67

# Getting “no space left on device” errors with Boot2Docker?

If you're using Boot2Docker with a large number of images, or the images you're
working with are very large, your pulls might start failing with "no space left 
on device" errors when the Boot2Docker volume fills up. There are two solutions
you can try.

## Solution 1: Add the `DiskImage` property in boot2docker profile

The `boot2docker` command reads its configuration from the `$BOOT2DOCKER_PROFILE` if set, or `$BOOT2DOCKER_DIR/profile` or `$HOME/.boot2docker/profile` (on Windows this is `%USERPROFILE%/.boot2docker/profile`).  

1. View the existing configuration, use the `boot2docker config` command.

        $ boot2docker config
        # boot2docker profile filename: /Users/mary/.boot2docker/profile
        Init = false
        Verbose = false
        Driver = "virtualbox"
        Clobber = true
        ForceUpgradeDownload = false
        SSH = "ssh"
        SSHGen = "ssh-keygen"
        SSHKey = "/Users/mary/.ssh/id_boot2docker"
        VM = "boot2docker-vm"
        Dir = "/Users/mary/.boot2docker"
        ISOURL = "https://api.github.com/repos/boot2docker/boot2docker/releases"
        ISO = "/Users/mary/.boot2docker/boot2docker.iso"
        DiskSize = 20000
        Memory = 2048
        CPUs = 8
        SSHPort = 2022
        DockerPort = 0
        HostIP = "192.168.59.3"
        DHCPIP = "192.168.59.99"
        NetMask = [255, 255, 255, 0]
        LowerIP = "192.168.59.103"
        UpperIP = "192.168.59.254"
        DHCPEnabled = true
        Serial = false
        SerialFile = "/Users/mary/.boot2docker/boot2docker-vm.sock"
        Waittime = 300
        Retries = 75

  The configuration shows you where `boot2docker` is looking for the `profile` file. It also output the settings that are in use. 


2. Initialise a default file to customize using `boot2docker config > ~/.boot2docker/profile` command.

3. Add the following lines to `$HOME/.boot2docker/profile`:

        # Disk image size in MB
        DiskSize = 50000

4. Run the following sequence of commands to restart Boot2Docker with the new settings.

        $ boot2docker poweroff
        $ boot2docker destroy
        $ boot2docker init
        $ boot2docker up

## Solution 2: Increase the size of boot2docker volume

This solution increases the volume size by first cloning it, then resizing it
using a disk partitioning tool. We recommend
[GParted](http://gparted.sourceforge.net/download.php/index.php). The tool comes
as a bootable ISO, is a free download, and works well with VirtualBox.

1. Stop Boot2Docker

  Issue the command to stop the Boot2Docker VM on the command line:

      $ boot2docker stop

2. Clone the VMDK image to a VDI image

  Boot2Docker ships with a VMDK image, which can’t be resized by VirtualBox’s 
  native tools. We will instead create a VDI volume and clone the VMDK volume to 
  it. 

3. Using the command line VirtualBox tools, clone the VMDK image to a VDI image:

        $ vboxmanage clonehd /full/path/to/boot2docker-hd.vmdk /full/path/to/<newVDIimage>.vdi --format VDI --variant Standard

4. Resize the VDI volume

  Choose a size that will be appropriate for your needs. If you’re spinning up a 
  lot of containers, or your containers are particularly large, larger will be 
  better:

      $ vboxmanage modifyhd /full/path/to/<newVDIimage>.vdi --resize <size in MB>

5. Download a disk partitioning tool ISO 

  To resize the volume, we'll use [GParted](http://gparted.sourceforge.net/download.php/). 
  Once you've downloaded the tool, add the ISO to the Boot2Docker VM IDE bus. 
  You might need to create the bus before you can add the ISO. 

  > **Note:** 
  > It's important that you choose a partitioning tool that is available as an ISO so 
  > that the Boot2Docker VM can be booted with it.

  <table>
      <tr>
          <td><img src="/articles/b2d_volume_images/add_new_controller.png"><br><br></td>
      </tr>
      <tr>
          <td><img src="/articles/b2d_volume_images/add_cd.png"></td>
      </tr>
  </table>

6. Add the new VDI image 

  In the settings for the Boot2Docker image in VirtualBox, remove the VMDK image 
  from the SATA controller and add the VDI image.

  <img src="/articles/b2d_volume_images/add_volume.png">

7. Verify the boot order

    In the **System** settings for the Boot2Docker VM, make sure that **CD/DVD** is 
    at the top of the **Boot Order** list.

    <img src="/articles/b2d_volume_images/boot_order.png">

8. Boot to the disk partitioning ISO

  Manually start the Boot2Docker VM in VirtualBox, and the disk partitioning ISO 
  should start up. Using GParted, choose the **GParted Live (default settings)** 
  option. Choose the default keyboard, language, and XWindows settings, and the 
  GParted tool will start up and display the VDI volume you created. Right click 
  on the VDI and choose **Resize/Move**. 

  <img src="/articles/b2d_volume_images/gparted.png">

9. Drag the slider representing the volume to the maximum available size.

10. Click **Resize/Move** followed by **Apply**. 

  <img src="/articles/b2d_volume_images/gparted2.png">

11. Quit GParted and shut down the VM. 

12. Remove the GParted ISO from the IDE controller for the Boot2Docker VM in
VirtualBox.

13. Start the Boot2Docker VM 

  Fire up the Boot2Docker VM manually in VirtualBox. The VM should log in 
  automatically, but if it doesn't, the credentials are `docker/tcuser`. Using 
  the `df -h` command, verify that your changes took effect.

  <img src="/articles/b2d_volume_images/verify.png">

You’re done!

thaJeztah commented:
@gustavohenrique oh, :cry: it looks like this PR needs a rebase, because the docs were moved from `/docs/sources/` to `/docs/` this document can now be found here: [/docs/articles/b2d_volume_resize.md](https://github.com/docker/docker/blob/master/docs/articles/b2d_volume_resize.md).

Are you able to rebase and update with the text that @moxiegirl provided, or would you like one of the doc maintainers to carry this for you?

Commit:
client/load: tiny typo fix

Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

LK4D4 committed:
Merge pull request #14383 from runcom/runconfig-merge-cleanup

add break after key is found in for loop

runcom committed:
Merge pull request #14374 from Mashimiao/clinent-load-tiny-typo-fix

client/load: tiny typo fix

gustavohenrique committed:
fixed the boot2docker text

Removal of the regex to replace ips

Signed-off-by: Mingzhen Feng <fmzhen@zju.edu.cn>
Signed-off-by: Gustavo Henrique <gustavo@gustavohenrique.net>

Use goroutine-safe version of rand.Source

Signed-off-by: Alexander Morozov <lk4d4@docker.com>
Signed-off-by: Gustavo Henrique <gustavo@gustavohenrique.net>

fixed the boot2docker text

gustavohenrique committed:
fixed the boot2docker text

gustavohenrique committed:
fixed the boot2docker text

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "patch-1" git@github.com:gustavohenrique/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~8
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

gustavohenrique committed:
fixed the boot2docker text

gustavohenrique commented:
@thaJeztah, I used the text provided by @moxiegirl.
Thanks @moxiegirl

moxiegirl commented:
Thanks @gustavohenrique much better. Before we can merge, please rebase and squash your commits. This page explains how: http://docs.docker.com/project/create-pr/#rebase-your-branch

If you can't do this happy to carry.

gustavohenrique committed:
Merge branch 'patch-1' of github.com:gustavohenrique/docker into patch-1

gustavohenrique commented:
@moxiegirl done! Can you check it, please?

moxiegirl commented:
@gustavohenrique Looks like it got all wonky.  Would you like me to carry this change @gustavohenrique?

gustavohenrique commented:
@moxiegirl Yes, please!
